pkg/loadbalancer/redirectpolicy/controller.go

```diff
@@ -374,6 +379,15 @@ func (c *lrpController) updateRedirectBackends(wtxn writer.WriteTxn, lrp *LocalR
 
 	}
 
+	// Function to compare whether the new BackendParams produced in the loop below
+	// is equal to the old one. We only compare subset of fields as some of the fields
+	// are managed by the load-balancing control-plane.
+	compareBackendParams := func(a, b *lb.BackendParams) bool {
+		return a.Address == b.Address &&
+			a.State == b.State &&
+			slices.Equal(a.PortNames, b.PortNames)
```

Comment:
We should compare more fields like `Unhealthy`? And potentially `Weight` in case users set the weight of a backend at runtime using the command line.